### PR TITLE
 build: define compile time features with buildflag header

### PR DESCRIPTION
**Pull request by deepak1556**
##### Description of Change

Also isolated the feature args for easier includes in other build targets.

##### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

##### Release Notes

Notes: no-notes

### CODE REVIEW

File: atom/app/atom_main.cc [hunk 1]
```diff
@@ -35,14 +35,14 @@
...
+#include "electron/buildflags/buildflags.h"
 
 namespace {
 
-#ifdef ENABLE_RUN_AS_NODE
+#if BUILDFLAG(ENABLE_RUN_AS_NODE)
 const char kRunAsNode[] = "ELECTRON_RUN_AS_NODE";
 #endif
 
-#if defined(ENABLE_RUN_AS_NODE) || defined(OS_WIN)
```
Comment (alexeykuzmin) [inline atom/app/atom_main.cc]: Why? )

Comment (deepak1556) [reply atom/app/atom_main.cc]: The method already has guards wrt implementation, also it seems to be generic, don't see the reason to specifically guard it behind `ELECTRON_RUN_AS_NODE`

File: atom/app/node_main.cc [hunk 2]
```diff
@@ -2,8 +2,6 @@
 // Use of this source code is governed by the MIT license that can be
 // found in the LICENSE file.
 
-#ifdef ENABLE_RUN_AS_NODE
```
Comment (alexeykuzmin) [inline atom/app/node_main.cc]: Should this file be excluded from build if we build without `ENABLE_RUN_AS_NODE `?

Comment (deepak1556) [reply atom/app/node_main.cc]: The files are already behind `enable_run_as_node` , hence I removed this guard.

File: buildflags/BUILD.gn [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  flags = [
+    "ENABLE_DESKTOP_CAPTURER=$enable_desktop_capturer",
+    "ENABLE_RUN_AS_NODE=$enable_run_as_node",
+    "ENABLE_OSR=$enable_osr",
+    "ENABLE_VIEW_API=$enable_view_api",
+    "ENABLE_PEPPER_FLASH=$enable_pepper_flash",
+    "ENABLE_PDF_VIEWER=$enable_pdf_viewer",
+    "OVERRIDE_LOCATION_PROVIDER=$enable_fake_location_provider",
+  ]
+}
```
Comment (nornagon) [inline buildflags/BUILD.gn]: Could this go in `build/` instead of in a new dir?

Comment (deepak1556) [reply buildflags/BUILD.gn]: I was planning to add it under `build/` but went with the directory structure upstream components use.

**Comment by jkleinsc** [comment]
Merging as the failing linux-x64-testing-tests was resolved via #14875

**Pull request merged**